v2/go.mod

```diff
@@ -0,0 +1,27 @@
+module github.com/ipld/go-car/v2
+
+go 1.15
+
+require (
+	github.com/google/go-cmp v0.5.5 // indirect
+	github.com/google/uuid v1.2.0 // indirect
+	github.com/ipfs/go-blockservice v0.1.4 // indirect
+	github.com/ipfs/go-cid v0.0.7
+	github.com/ipfs/go-datastore v0.4.5 // indirect
+	github.com/ipfs/go-ipld-cbor v0.0.5
+	github.com/ipfs/go-ipld-format v0.2.0
+	github.com/ipfs/go-log v1.0.5 // indirect
+	github.com/ipld/go-car v0.3.1
+	github.com/ipld/go-ipld-prime v0.10.0 // indirect
+	github.com/klauspost/cpuid/v2 v2.0.6 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
+	github.com/stretchr/testify v1.7.0
+	github.com/whyrusleeping/cbor-gen v0.0.0-20210422071115-ad5b82622e0f // indirect
+	github.com/willscott/carbs v0.0.5-0.20210604075022-1036c89e1fd7
+	go.uber.org/multierr v1.7.0 // indirect
+	go.uber.org/zap v1.17.0 // indirect
+	golang.org/x/crypto v0.0.0-20210513164829-c07d793c2f9a // indirect
+	golang.org/x/sys v0.0.0-20210608053332-aa57babbf139 // indirect
+)
+
+replace github.com/ipld/go-car => ../
```

Comment:
have you given much thought to whether Go's v2 module should depend on the v1 module, or if it should stand on its own ground?

Having the dependency has the upside of less code duplication. the v1 module works, and if the APIs are going to be largely compatible, it's nice to just do it that way.

There are two main downsides I see to having v2 depend on v1: we'll now have to maintain two modules instead of one (e.g. coordinating v1 and then v2 releases), and if v2 needs to directly use any internal code from v1, you won't be able to given they are in different packages.

Reply:
If the aim is to minimise barrier for adoption, then I _think_ dependency is the path of least resistance. Open to what you folks think works best.

Reply:
If that were the case, then I would assume we would not be doing a v2 Go module at all - we would just add new APIs or options to the v1 Go module.

From what Will said, I understood that we'd break away from the v1 module and do a v2 module, while keeping APIs that work with the old IPLD libraries to make upgrading a bit easier. upgrading then would, ideally, just mean adding v2 to the go-car import paths and doing small local tweaks.

It seems to me like what you're doing here is a middle ground, which kinda gets us the disadvantages of both approaches at the same time :) people on the v1 module will still have to edit their code to add support for the v2 module, and on our side we'll add work for ourselves by maintaining two modules instead of one.

Reply:
After a brief chat on Slack, I realised I wasn't thinking about existing carv1 users. They should continue to work with zero changes, ideally, which means we'd have to continue maintaining the v1 module anyway. So I'm leaning towards my comments above being wrong.

Reply:
Thank you for explaining; that makes sense.

The approach in this PR assumes that "small local tweaks" is not acceptable on the v1 users' codebase. The v2 would then return types from the v1 library, as well as providing a newer IPLD flavoured library. 

If we can get away with enforcing small local tweaks to users whom want to use v2 then great, I think what you are suggesting with a totally separate v2 module is much cleaner :)

@willscott what are your thoughts?